src/check_executor.rs

```diff
@@ -80,6 +83,15 @@ impl CheckSender {
 
         resolve_rx
     }
+
+    /// Requeues the check to be executed again, increasing the number of retries by 1
+    fn queue_check_for_retry(&self, mut check: ScheduledCheck) {
+        check.retry_count += 1;
+        self.queue_size.fetch_add(1, Ordering::SeqCst);
+        self.sender
+            .send(check)
+            .expect("Failed to queue ScheduledCheck");
+    }
```

Comment:
At some point we might want to add some kind of wrapper around this queue so that we're not duplicating code

Reply:
I think this is OK actually, where it's located _is_ the wrapper